workflow/serializers.py

```
@@ -129,6 +130,31 @@ class CoreUserInvitationSerializer(serializers.Serializer):
                                    min_length=1, max_length=10)
 
 
+class CoreUserResetPasswordSerializer(serializers.Serializer):
+    """
+    Serializer for reset password request data, it use built-in django form for password resetting
+    to validate the fields. We override only send_mail method of this form.
+    """
+
+    email = serializers.EmailField()
+
+    def validate_email(self, value):
+        self._reset_form = CoreUserPasswordResetForm(data=self.initial_data)
+        if not self._reset_form.is_valid():
+            raise serializers.ValidationError(self._reset_form.errors)
+        return value
+
+    def save(self):
+        assert hasattr(self, '_reset_form'), "You should validate serializer before saving"
```

I think it's better to raise an error different of `AssertionError`